File: server/FitMate.Core/JsonModels/ProgramPlans/ProgramPlanScheduleRuleModel.cs
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.ProgramPlans;

public class ProgramPlanScheduleRuleModel
{
    public long Id { get; set; }
    public DayOfWeek? DayOfWeek { get; set; }
    public int? RotationDayIndex { get; set; }
    public ProgramPlanDayType DayType { get; set; }
    public long? WorkoutTemplateId { get; set; }
    public string? WorkoutTemplateName { get; set; }
    public int WeekInterval { get; set; }
    public int OrderIndex { get; set; }
    public bool IsOptional { get; set; }
}

File: client/src/types/JsonModels/ProgramPlans/ProgramPlanScheduleRuleModel.ts
import type { DayOfWeek } from "../Enums/DayOfWeek";
import type { ProgramPlanDayType } from "../Enums/ProgramPlanDayType";

export interface ProgramPlanScheduleRuleModel
{
	id: number;
	dayOfWeek?: DayOfWeek;
	rotationDayIndex?: number;
	dayType: ProgramPlanDayType;
	workoutTemplateId?: number;
	workoutTemplateName?: string;
	weekInterval: number;
	orderIndex: number;
	isOptional: boolean;
}
